docs/proposals/container-runtime-interface/capabilities.md

```diff
@@ -0,0 +1,134 @@
+is the final capability set).
+
+## Solutions
+
+As mentioned above, an explicit whitelist type of capability-retain-set could make things very consistent across runtimes and
+version updates.
+But as we can see, it's not very practical for system admins and operators to put a long [default capability list](https://docs.docker.com/engine/reference/run/#/runtime-privilege-and-linux-capabilities) on every
+container he wants to launch.
+Besides, changing the k8s API could take several release cycles.
+
+So in the proposed solution here. We will:
+- Keep the k8s API as is today. Users will not need to change anything.
+- Define a default capability list inside k8s, and document about it.
+- Make CRI expose the whitelist based capability interface, and put the burden on the runtime to implement that.
+- Make the final capability set **visible** through the container status, for example:
+
+```shell
+$ kubectl get pod -o yaml
+apiVersion: v1
+items:
+- apiVersion: v1
+  kind: Pod
+  metadata:
+  ...
+  spec:
+    containers:
+    - command:
+      - sleep
+      - "1000"
+      image: busybox
+      imagePullPolicy: Always
+      name: my-busybox
+      securityContext:
+        capabilities:
+          add:
+          - NET_ADMIN
+      ...
+  status:
+    containerStatuses:
+    - image: busybox
+    capabilities:
+      SETPCAP, MKNOD, AUDIT_WRITE, ...
+      name: my-busybox
+      state:
+        running:
+          startedAt: 2016-09-27T22:33:38Z
+    ...
+```
+
+## Open questions:
+
+**Q:** What the default capability list should be?
+
+**A:** According to @smarterclayton, there has been questions and arguments about what the default capability list should be, and it's not settled down yet.
+       So for now, in order to keep backward compatibility, we'd better just define it as [Docker's default capability list](https://docs.docker.com/engine/reference/run/#/runtime-privilege-and-linux-capabilities) today.
+
+**Q:** How wide does the default capability list apply? Does it apply to one node, or the whole cluster?
+
+**A:** I don't really have a use case for support per-node-default default capability list, if someone could give one, it might be very helpful.
+       Other than that, for today's use case, a cluster-wide default capability list is enough.
+
+**Q:** How does runtimes implement the whitelist capability set?
+
+**A:** For Docker runtime, this adds a burden to Docker runtime maintainers to convert the capabilities in the whitelist to `Add` and `Drop` set
+       based on the Docker's default capability list.
+       For rkt runtime, this can be done by using the `--caps-retain` flag.
+
+## Changes need to make:
+
+- Add a `DefaultCapabilities` field in the [PodSecurityPolicySpec](https://github.com/kubernetes/kubernetes/blob/v1.5.0-alpha.0/pkg/apis/extensions/types.go#L653), and make it default to today's default capability list.
```

Comment:
I'm not sure I understand the goal of this change.  The `DefaultAddCapabilities` of PSP sound like what you're shooting for here, caps that are always added to a pod unless specifically requested to be dropped in the pod spec.  Also, the `AllowedCapabilities` field acts as a white list of things that may be added but aren't automatically added.